src/Avalonia.Controls/TextBlock.cs

```diff
@@ -708,7 +708,7 @@ protected override Size MeasureOverride(Size availableSize)
                 //Reset TextLayout when the constraint is not matching.
                 _textLayout?.Dispose();
                 _textLayout = null;
-                _constraint = deflatedSize;
+                _constraint = new Size(double.PositiveInfinity, deflatedSize.Height);
```

Comment:
This doesn't seem right, we shouldn't blindly ignore the available size.
It will break any control depending on its children desired size, effectively making the measure pass useless in this case.

e.g. the following will break (no more text wrapping):

```xaml
<StackPanel>
  <TextBlock
    Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    TextWrapping="Wrap" />
</StackPanel>
```

I believe that we should really add more text block measure tests with any new changes.

Reply:
Yes you are right. This will break text wrapping in some situations. Time to add more tests.

Reply:
These are the test cases I came up with: 
```
 <StackPanel>
   <StackPanel
       HorizontalAlignment="Center"
       Width="200"
       Background="LightBlue">
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Right" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Right" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Right" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Center" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Center" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Center" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Left" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Left" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Background="Red" HorizontalAlignment="Left" TextAlignment="Right"/>
   </StackPanel>
   <StackPanel
       Margin="0 4 0 0"
       HorizontalAlignment="Center"
       Width="200"
       Background="LightBlue">
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Right" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Right" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Right" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Center" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Center" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Center" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Left" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Left" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="150" Background="Red" HorizontalAlignment="Left" TextAlignment="Right"/>
   </StackPanel>
   <StackPanel
       Margin="0 4 0 0"
       HorizontalAlignment="Center"
       Width="200"
       Background="LightBlue">
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Right"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Left"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Center"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Right"/>
   </StackPanel>
   <StackPanel
       Margin="0 4 0 0"
       HorizontalAlignment="Center"
       Width="200"
       Background="LightBlue">
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Left" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Center" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Right" TextAlignment="Right" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Left" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Center" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Center" TextAlignment="Right" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Left" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Center" TextWrapping="Wrap"/>
     <TextBlock Text="Hello World" Width="44" Background="Red" HorizontalAlignment="Left" TextAlignment="Right" TextWrapping="Wrap"/>
   </StackPanel>
 </StackPanel>
```